frontend/src/lib/components/AnnotationsOverlay/AnnotationsOverlay.tsx

```diff
@@ -0,0 +1,248 @@
+}
+
+interface AnnotationsBadgeCSSProperties extends React.CSSProperties {
+    '--annotations-badge-index': number
+    '--annotations-badge-scale': number
+}
+
+const AnnotationsBadge = React.memo(function AnnotationsBadgeRaw({ index, date }: AnnotationsBadgeProps): JSX.Element {
+    const { intervalUnit, groupedAnnotations, isDateLocked, activeDate, isPopoverShown } =
+        useValues(annotationsOverlayLogic)
+    const { activateDate, deactivateDate, lockDate, unlockDate } = useActions(annotationsOverlayLogic)
+
+    const [hovered, setHovered] = useState(false)
+    const buttonRef = useRef<HTMLButtonElement>(null)
+
+    const dateGroup = determineAnnotationsDateGroup(date, intervalUnit)
+    const annotations = groupedAnnotations[dateGroup] || []
+
+    const active = activeDate === date && isPopoverShown
+    const shown = active || hovered || annotations.length > 0
+
+    return (
+        <button
+            ref={buttonRef}
+            className="AnnotationsBadge"
+            // eslint-disable-next-line react/forbid-dom-props
+            style={
+                {
+                    '--annotations-badge-index': index,
+                    '--annotations-badge-scale': shown ? 1 : 0,
+                } as AnnotationsBadgeCSSProperties
+            }
+            onMouseEnter={() => {
+                setHovered(true)
+                console.log(activeDate?.valueOf(), date.valueOf(), activeDate === date)
+                if (!isDateLocked) {
+                    const button = buttonRef.current as HTMLButtonElement
+                    activateDate(date, [button.offsetLeft, button.offsetTop])
+                }
+            }}
+            onMouseLeave={() => {
+                setHovered(false)
+                console.log(activeDate?.toISOString(), date.toISOString(), activeDate === date)
+                if (!isDateLocked) {
+                    deactivateDate()
+                }
+            }}
+            onClick={
+                !isDateLocked
+                    ? lockDate
+                    : active
+                    ? unlockDate
+                    : () => {
+                          const button = buttonRef.current as HTMLButtonElement
+                          activateDate(date, [button.offsetLeft, button.offsetTop])
+                      }
+            }
+        >
+            <LemonBubble
+                count={annotations.length || <IconPlusMini className="w-full h-full" />}
+                size="small"
+                style={active && isDateLocked ? { outline: '0.125rem solid var(--primary)' } : undefined}
+            />
+        </button>
+    )
+})
+
+function AnnotationsPopover(): JSX.Element {
+    const { popoverAnnotations, activeDate, intervalUnit, isDateLocked, insightId } = useValues(annotationsOverlayLogic)
+    const { closePopover } = useActions(annotationsOverlayLogic)
+    const { openModalToCreateAnnotation } = useActions(annotationModalLogic)
+
+    return (
+        <LemonModal
```

Comment:
Why not wrap this with a div called AnnotationsPopover?

Goes against some of our Design System principles to be modifying the standard component as now it is hard to reason on the behaviour of the LemonModal.

Reply:
Fair enough, restructured this to minimize affecting `LemonModal` directly.